### PR TITLE
Enhance the non-PII telemetry capture to improve the user experience

### PR DESCRIPTION
**Pull request by samaea**
## PR Summary

Addresses this issue:- https://github.com/Azure/AKS-Construction/issues/521.

## PR Checklist

- [X] PR has a meaningful title
- [X] Summarized changes
- [X] This PR is ready to merge and is not **Work in Progress**
- [X] Link to a filed issue
- [X] Screenshot of UI changes (if PR includes UI changes)

image

### CODE REVIEW

**Review by Gordonby — APPROVED** [review]

File: helper/src/components/portalnav.js [hunk 1]
```diff
@@ -109,13 +109,16 @@ export default function PortalNav({ config }) {
   const { tabLabels, defaults, presets } = config
   const [pivotkey, setPivotkey] = useState(Object.keys(tabLabels)[0])
   useAITracking("PortalNav", tabLabels[pivotkey])
-
   const [urlParams, setUrlParams] = useState(new URLSearchParams(window.location.search))
   const [invalidArray, setInvalidArray] = useState(() => Object.keys(defaults).reduce((a, c) => { return { ...a, [c]: [] } }, {}))
+  useAITracking("PageNav", urlParams.get('preset') || 'defaultOps')
```
Comment (khowling) [inline helper/src/components/portalnav.js:114]: I'd like to see `useAITracking` here trigger on the `selected` state rarther than use the url.

so `useAITracking('PageNav', selected.preset)`

File: helper/src/components/portalnav.js [hunk 2]
```diff
@@ -109,13 +109,16 @@ export default function PortalNav({ config }) {
...
   // The selected cards within the sections for the chosen preset, for example { "ops": "normal", "secure": "high" }
   const [selected, setSelected] = useState(initSelected(urlParams.get('preset') || 'defaultOps'))
   // The tabValues, for example { "deploy": { "clusterName": "az234"}}
   const [tabValues, setTabValues] = useState(initTabValues(selected, defaults, true))
+  // Field Selections - Used to keep track of the last FieldSelections monitored by App Insights to prevent logging the same entry continuously
+  const [tabState, setTab] = useState("");
+  const [fieldState, setField] = useState("");
```
Comment (khowling) [inline helper/src/components/portalnav.js:121]: Lets discuss

File: helper/src/components/portalnav.js [hunk 3]
```diff
@@ -195,8 +198,8 @@ export default function PortalNav({ config }) {
   }
 
   function updateSelected(sectionKey, cardKey) {
-    console.log("Update Selected Fired " + sectionKey + " - " + cardKey)
-
+    console.log("AI:- Card update fired " + sectionKey + " - " + cardKey)
```
Comment (khowling) [inline helper/src/components/portalnav.js:201]: remove the log?

File: helper/src/components/portalnav.js [hunk 4]
```diff
@@ -195,8 +198,8 @@ export default function PortalNav({ config }) {
...
+    appInsights.trackEvent({name: "Card." + sectionKey + "." + cardKey});
```
Comment (khowling) [inline helper/src/components/portalnav.js:202]: very good!!

File: helper/src/components/portalnav.js [hunk 5]
```diff
@@ -282,9 +285,15 @@ export default function PortalNav({ config }) {
   }
 
   function mergeState(tab, field, value, previewLink) {
-
     let updatevals
     let newFields = new Map()
+    if(tabState !== tab || fieldState !== field){
```
Comment (khowling) [inline helper/src/components/portalnav.js:290]: lets discuss

Comment (samaea) [reply helper/src/components/portalnav.js]: Resolved by combining both variables into one state object. Thanks for the feedback @khowling! :)

File: helper/src/components/portalnav.js [hunk 6]
```diff
@@ -295,6 +301,7 @@ export default function PortalNav({ config }) {
       }
     }
 
+    setLastAIUpdated({tab: tab, field: field})
```
Comment (khowling) [inline helper/src/components/portalnav.js:304]: Hey Sam,  just one optimisation, you can put this in the body of your new if statement no?  so if the tab/field hasn't changed, you dont need to update the state?

**Review by khowling — APPROVED** [review]
thanks for incorporating thoes changes, look good

**Review by khowling — APPROVED** [review]

**Pull request merged**